zkstack_cli/crates/common/src/server.rs

```diff
@@ -99,3 +106,120 @@ impl Server {
         })
     }
 }
+
+#[allow(clippy::too_many_arguments)]
+async fn run_server<P>(
+    shell: &Shell,
+    uring: Option<&str>,
+    chains_folder: P,
+    genesis_path: P,
+    wallets_path: P,
+    general_path: P,
+    secrets_path: P,
+    contracts_path: P,
+    additional_args: Vec<String>,
+    execution_mode: ExecutionMode,
+    server_mode: ServerMode,
+    tag: Option<String>,
+) -> anyhow::Result<()>
+where
+    P: AsRef<OsStr>,
+{
+    let mut cmd = match execution_mode {
+        ExecutionMode::Release => cargo_run(
+            shell,
+            true,
+            uring,
+            genesis_path,
+            wallets_path,
+            general_path,
+            secrets_path,
+            contracts_path,
+            additional_args,
+        ),
+        ExecutionMode::Debug => cargo_run(
+            shell,
+            false,
+            uring,
+            genesis_path,
+            wallets_path,
+            general_path,
+            secrets_path,
+            contracts_path,
+            additional_args,
+        ),
+        ExecutionMode::Docker => {
+            let tag = tag.unwrap_or(select_tag().await?.to_owned());
+
+            Cmd::new(cmd!(
+                shell,
+                "docker run
+                --platform linux/amd64
+                --net=host
+                -v {chains_folder}:/chains
```

Comment:
You have to have fixed paths inside container and have volumes for mounting the exact files

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3314/commits/debb5070631f6669b5081cd829980f78f1ac062d